rose/server/score.py

```diff
@@ -27,54 +27,64 @@ def process(players, track):
             if player.x < config.matrix_width - 1:
                 player.x += 1
 
-    # Now handle obstacles, preferring players in their own lane.
+        obstacle = track.get(player.x, player.y)
 
-    sorted_players = sorted(six.itervalues(players),
-                            key=lambda p: 0 if p.in_lane() else 1)
-    positions = set()
+        if obstacle == obstacles.NONE:
+            # Move forlward
+            player.score += config.score_move_forward
 
-    for player in sorted_players:
-        player.score += config.score_move_forward
-        obstacle = track.get(player.x, player.y)
-        if obstacle == obstacles.CRACK:
-            if player.action != actions.JUMP:
-                track.clear(player.x, player.y)
-                player.y += 1
-                player.score += config.score_move_backward * 2
-            else:
-                player.score += config.score_jump
         elif obstacle in (obstacles.TRASH,
                           obstacles.BIKE,
                           obstacles.BARRIER):
-            if player.action not in (actions.LEFT, actions.RIGHT):
+            # Move back
+            track.clear(player.x, player.y)
+            player.y += 1
+            player.score += config.score_move_backward
+
+        elif obstacle == obstacles.CRACK:
+            if player.action == actions.JUMP:
+                # Move forlward
+                player.score += config.score_move_forward
+                player.score += config.score_jump
```

Comment:
We can add the score with one line:

     player.score += config.score_move_forward + config.score_jump

Reply:
done